lib/spack/spack/test/util/executable.py

```diff
@@ -36,22 +35,36 @@ def test_read_unicode(tmpdir, working_env):
         filter_shebangs_in_directory('.', [script_name])
 
         # read the unicode back in and see whether things work
-        script = ex.Executable('./%s' % script_name)
+        if sys.platform == 'win32':
+            script = ex.Executable('%s %s' % (sys.executable, script_name))
+        else:
+            script = ex.Executable('./%s' % script_name)
+
         assert u'\xc3' == script(output=str).strip()
 
 
 def test_which_relative_path_with_slash(tmpdir, working_env):
-    tmpdir.ensure('exe')
-    path = str(tmpdir.join('exe'))
+    tmpdir.ensure("exe")
+    path = str(tmpdir.join("exe"))
+
+    
     os.environ['PATH'] = ''
 
     with tmpdir.as_cwd():
-        no_exe = ex.which('./exe')
+        no_exe = ex.which('.{0}exe'.format(os.path.sep))
         assert no_exe is None
 
-        fs.set_executable(path)
-        exe = ex.which('./exe')
-        assert exe.path == path
+
+        with tmpdir.as_cwd():
```

Comment:
Instead of using `fs.touch()` here, add an `if sys.platform == "win32":` above and use `tmpdir.ensure('exe.exe')` instead

Reply:
Should I just remove the else clause and make the if path perform the the non windows functions (touch and set_exec)? This simplifies the code a bit.

Reply:
Do you mean at the beginning of the function?

Reply:
I'm thinking at the beginning of the function:

> if sys.platform == "win32":
>     tmpdir.ensure('exe.exe')
> else:
>     tmpdir.ensure('exe')